mods/page_revisions.js

```diff
@@ -424,6 +558,105 @@ PRS.prototype.getTitleRevision = function(restbase, req) {
     // TODO: handle other revision formats (tid)
 };
 
+PRS.prototype._createRenameChecker = function(restbase, req) {
+    var self = this;
+    var rp = req.params;
+    return function(res) {
+        if (res.pageData
+                && res.pageData.body.items
+                && res.pageData.body.items.length) {
+            var latestEvent = res.pageData.body.items[0];
+            var latestRev = res.revisionInfo.body.items[0];
+            if (latestEvent.event_type === 'rename_to'
+                    && uuid.fromString(latestEvent.tid).getDate()
+                        >= uuid.fromString(latestRev.tid).getDate()) {
+                return self._getLatestPageTitle(restbase, req, latestEvent)
+                .then(function(latestTitle) {
+                    var rootPath = restbase._rootReq.uri.path;
+                    var newPath = [];
+                    rootPath.forEach(function(pathElement) {
+                        if (pathElement === rp.title) {
+                            newPath.push(encodeURIComponent(latestTitle));
+                        } else {
+                            newPath.push(pathElement);
+                        }
+                    });
+                    throw new rbUtil.HTTPRedirect({
```

Comment:
I don't think it will, honestly. Maybe we could use a relative redirect instead?